containers/daap-python-base/src/var/task/data_product_metadata.py

```diff
@@ -1,116 +1,326 @@
 import json
 import traceback
+from copy import deepcopy
+from typing import Dict
 
 import boto3
 import botocore
 from data_platform_logging import DataPlatformLogger
-from data_platform_paths import DataProductConfig
+from data_platform_paths import (
+    BucketPath,
+    DataProductConfig,
+    JsonSchemaName,
+    specification_path,
+    specification_prefix,
+)
 from dataengineeringutils3.s3 import get_filepaths_from_s3_folder, read_json_from_s3
 from jsonschema import validate
 from jsonschema.exceptions import ValidationError
 
 s3_client = boto3.client("s3")
 
+glue_csv_table_input_template = {
```

Comment:
I think this is alright for now, but @tom-webber was floating the idea earlier of having another layer in our codebase for modelling the schema and metadata structure, so that we can ensure that the representations are consistent and valid against our JSON schemas. It's hard to plan out while we are working on different endpoints in parallel, but maybe we can iterate on this once we have an initial version of each one.

The way I see it, we have two representations that we want to transform back and forth between: the one we have specced out in JSON schema, and the simpler one we are going to expose through the API. I'm curious if we can make use of something like this in powertools: https://docs.powertools.aws.dev/lambda/python/latest/utilities/parser/

Reply:
yeah that sounds like a good idea to me